### ISSUE TITLE
[Accessibility] Narrator/NVDA is not reading the unchecked status for the list item 

### ISSUE THREAD

**Issue by setaskin** [microsoft/vscode#120597]
- VS Code Version:
- OS Version:

Steps to Reproduce:

1. go to explorer pane 
2. click on tree dots in the top right of the tree view.
3. uncheck couple items in the list ( leave some of them as checked)
3. open Narrator app 
4. navigate trough same list and see Narrator reads "unchecked" status.

Actual :  Narrator only reads "checked" status of the item. If the item is "unchecked" narrator doesn't read the status 
Expected: Narrator should read the status as "unchecked" for unchecked items as well. 

![image](https://user-images.githubusercontent.com/37163639/113643755-e0928b00-9637-11eb-8d9d-d526108f3d5e.png)

Does this issue occur when all extensions are disabled?: Yes

**Comment by setaskin** [comment]
@isidorn , the bug is assigned to us from Accessibility team but I believe root cause is coming from vs code.

**Comment by isidorn** [comment]
I believe this is as designed behavior by Narrator. Narrator simply decides to not read out the when the item is unchecked.
@sbatten I see you nicely set the `aria-checked` [here](https://github.com/microsoft/vscode/blob/main/src/vs/base/browser/ui/menu/menu.ts#L650)

@setaskin do you face the same issue with NVDA? How about if you go the the breakpoints view and you uncheck a breakpoint?

**Comment by setaskin** [comment]
@isidorn , yes, NVDA also doesn't read "unchecked" status. but there is no problem on break points it reads that "breakpoint is added" or "breakpoint is removed".

**Comment by isidorn** [comment]
@setaskin thanks.
@sbatten I just checked this and I believe that you do not put `aria-checked=false` for unchecked items in the custom menu. Can you please double check this? Thanks

**Comment by sbatten** [comment]
@isidorn I think what's happening is that when the menu item is not checked, we set the `aria-checked` value to `false` but we also set the role to `menuitem` which probably tells the screen reader that `aria-checked` is not important.

I don't think we know if an action is toggleable, we just know if it is checked. I will have to update the action to be aware that it is unchecked an can be checked to label it properly

**Comment by isidorn** [comment]
@sbatten good point. They should have a role "checkbox" if they can be checked off. We do the same for the breakpoints and the list should support this.
Just an example here's how I do it for breakpoints https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/browser/breakpointsView.ts#L849

**Comment by RichCaloggero** [comment]
## Views not showing checked status via NVDA

Note that as long as at least one state is shown (preferrably checked), then not necessary to show the other, although would be nice to confirm it's a boolean.


- explorer view
- output view
- search view
- run view
- SCM CTRL view
- debug console
- terminal
- word wrap
Views / modes that do report checked status:

- minimap
- show breadcrumbs
- render whitespace
- render control characters

**Comment by RichCaloggero** [comment]
Can't seem to edit comments here...

All the views mentioned in previous comment refer to entries in the global view menu.

**Comment by isidorn** [comment]
@RichCaloggero thanks. I think we have found the root cause of the issue and we plan to look into it.
@sbatten let me know if I can help here.

**Comment by isidorn** [comment]
@sbatten ok I looked at the code and now I understand fully what you meant, you need to know if an action is `toggleable` as you said.
I have just verified with NVDA that this would indeed fix the issue, so we need to set `menuitemcheckbox` only for items that are checkable.

**Comment by setaskin** [comment]
Hi @isidorn , the accessibility team wants us to give fix date estimate on this ? Do you have an approx.. estimate when would it be fixed  ?

**Comment by Ashwinipola26** [comment]
@isidorn Could you please help us with the estimate date to fix this issue

**Comment by isidorn** [comment]
This is currently assigned to September which means we would like to fix it in September. Though it might get moved to October.
So rough estimate: in the next couple of months.

**Comment by fsteffi** [comment]
GitHubTags:#A11yTCS;#Win32;#DesktopApp;#A11y_VSCode;#WCAG4.2.2;#A11ySev2;#CAITestPass_May2020;#Visual Studio Code;#NVDA;#A11yMAS;#Narrator;
